packages/main/src/TabContainer.js

```diff
@@ -393,6 +399,8 @@ class TabContainer extends UI5Element {
 			};
 			item._itemSelectCallback = this._onItemSelect.bind(this);
 			item._getRealDomRef = () => this.getDomRef().querySelector(`*[data-ui5-stable=${item.stableDomRef}]`);
+			item._selectedTab = this._selectedTab;
```

Comment:
onBeforeRendering is growing quite large, 
better extract this whole filter fn into a seperate function.